src/handler.ts

```diff
@@ -34,22 +39,29 @@ export class CodestralHandler extends ChatModel {
     };
     this.messageAdded(msg);
     this._history.messages.push(msg);
-    const response = await this._mistralClient.chat({
-      model: 'codestral-latest',
-      messages: this._history.messages.map(msg => {
-        return {
-          role: msg.sender.username === 'User' ? 'user' : 'assistant',
-          content: msg.body
-        };
+    // const response = await this._mistralClient.chat({
+    //   model: 'codestral-latest',
+    //   messages: this._history.messages.map(msg => {
+    //     return {
+    //       role: msg.sender === 'User' ? 'user' : 'assistant',
+    //       content: msg.body
+    //     };
+    //   })
+    // });
+    const messages = mergeMessageRuns(
+      this._history.messages.map(msg => {
+        if (msg.sender.username === 'User') {
+          return new HumanMessage(msg.body);
+        }
+        return new SystemMessage(msg.body);
```

Comment:
Do we need to stack all the previous messages ?

I see that it was already the case before, but I wonder if it is necessary, it can become huge for a long conversation.

Reply:
Yes it can also increase the payload for the requests, but it gives the full context of the conversation as there is otherwise no way to have some kind of "memory"?

But maybe we can try using some higher level primitives offered by `langgraph` to hide this logic: https://js.langchain.com/docs/how_to/message_history/

Reply:
Also it looks like this `SystemMessage` should be switched to a `AIMessage`: https://js.langchain.com/docs/concepts/#aimessage

Reply:
> Yes it can also increase the payload for the requests, but it gives the full context of the conversation as there is otherwise no way to have some kind of "memory"?

Right, I was wondering if there could be some kind of "session" when using the API, and this history to be stored on the server providing the API.

> But maybe we can try using some higher level primitives offered by `langgraph` to hide this logic: https://js.langchain.com/docs/how_to/message_history/

We can probably keep it that way for a first PR, and think on how to improve the history in future issues / PRs.

Reply:
> Also it looks like this `SystemMessage` should be switched to a `AIMessage`: https://js.langchain.com/docs/concepts/#aimessage

Good catch.
Also the bot messages were not added to the handler history, nor to the API messages.